### ISSUE TITLE
how to initial workflow when i create new database record? ("Call to a member function getTos() on null" Error)

### ISSUE THREAD

**Issue by sjavadez** [zerodahero/laravel-workflow#4]
Hi
when I try to use a workflow, I get an error like this:
Call to a member function getTos() on null

my config/workflow.php file :
```
<?php

return [
    'straight'   => [
        'type'          => 'workflow',
        'marking_store' => [
            'type'      => 'multiple_state',
            'arguments' => ['state']
        ],
        'supports'      => ["App\Task"],
        'places'        => ['draft', 'review', 'publish'],
        'transitions'   => [
            'drafting' => [
                'from' => 'draft',
                'to'   => 'review',
            ],
            'reviewing' => [
                'from' => 'review',
                'to'   => 'publish',
            ]
        ],
    ]
];
```

my Controller :
```
public function store(Request $request)
    {
        // validate the given request
        $data = $this->validate($request, [
            'title' => 'required|string|max:255',
        ]);



        // create a new incomplete task with the given title
        $task = Auth::user()->tasks()->create([
            'title' => $data['title'],
        ]);

        $task->workflow_apply('drafting');

        // flash a success message to the session
        session()->flash('status', 'Task Created!');

        // redirect to tasks index
        return redirect('/tasks');
    }
```
how i can initialize state of my task when i create new task?

**Comment by zerodahero** [comment]
The Symfony Workflow component itself will treat `null` as the initial
place. Can you post your stack trace on that exception, and I can take a
look at it?

**Comment by sjavadez** [comment]
Yes , sure.

![screencapture-localhost-8000-tasks-2019-06-10-09_50_16](https://user-images.githubusercontent.com/27066228/59174134-5ffec280-8b65-11e9-9dd5-5407403f8d22.png)

**Comment by sjavadez** [comment]
When i Create new Task(my Model) and set Task's state , I get this error :

![screencapture-localhost-8000-tasks-2019-06-10-10_00_11](https://user-images.githubusercontent.com/27066228/59174480-c801d880-8b66-11e9-996e-eed53ff24431.png)

my Controller :
```
public function store(Request $request)
    {
        // validate the given request
        $data = $this->validate($request, [
            'title' => 'required|string|max:255',
        ]);



        // create a new incomplete task with the given title and state
        $task = Auth::user()->tasks()->create ([
            'title' => $data['title'] ,
            'state' => 'draft'
        ]);

        $task->workflow_can(['reviewing']); // False


        // flash a success message to the session
        session()->flash('status', 'Task Created!');

        // redirect to tasks index
        return redirect('/tasks');
    }
```

**Comment by zerodahero** [comment]
Ok, so as that exception states--the workflow component is expecting the workflow marking (looks like "state" on your `App\Task` model) to be an array. In your workflow config, you have the marking_store type set to "multiple_state", which requires that marking to be an array.

You have two options to address this:
1.) If this is your full workflow definition--it doesn't look like there's any need to have multiple states--single state would work fine, and you also don't have any transitions from a single initial place to multiple places, so you're only setup to be in one place at one time.
2.) If you do want to be in multiple places, add the array cast onto your marking:
```php
// on your Task model:
protected $casts = [
    "state" => "array"
];
```
You can read more about casting to an array on eloquent models in the Laravel docs: https://laravel.com/docs/5.8/eloquent-mutators#array-and-json-casting

**Comment by zerodahero** [comment]
For the first issue, looks like Symfony 4.2.9 fires the entered event on initialization. It has been fixed in version 2.0.2
